File: new/739_daily_temperatures_brute_force.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        // idea: brute force
        // find temp > curr temp in the given list, if not found add 0. 
        // track how many days it take to reach temp > curr temp from the curr day. 
        int n = temperatures.size();
        vector<int> until(n);
        for (int i = 0; i < temperatures.size(); i++) {
            int j = i + 1;
            int count = 1;
            while (j < n) {
                if (temperatures[j] > temperatures[i]) {
                    break;
                }
                count++;
                j++;
            }
            until[i] = (j == n) ? 0 : count;
        }

        return until;
    }
    // time: O(n^2)
    // space: O(n)
};

File: new/python/739_daily_temperatures_brute_force.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        # idea: brute force
        # +1 until we have a day greater than current day,
        # if we reach the end before getting a higher temp than current day
        # set the count to 0.
        until = []
        for i in range(len(temperatures)):
            j = i + 1
            count = 1
            while j < len(temperatures):
                if temperatures[i] < temperatures[j]:
                    break
                count += 1
                j += 1
            if j == len(temperatures):
                until.append(0)
            else:
                until.append(count)
        return until
        # time: O(n^2)
        # space: O(n)
